tests/manage/pv_services/test_rwo_pvc_fencing_unfencing.py

```diff
@@ -0,0 +1,632 @@
+        """
+        app_pod_nodes_name = []
+        # Todo: 3AZ Colocated scenario - select nodes from AZ having more than one worker node
+        if scenario == "colocated":
+            if ignore_check:
+                app_pod_nodes_name += ocs_nodes[0:num_of_nodes]
+            else:
+                mon_pods = pod.get_mon_pods()
+                mon_pod_nodes_name = [pod.get_pod_node(pod_obj).name for pod_obj in mon_pods]
+                osd_pods = pod.get_osd_pods()
+                osd_pod_nodes_name = [pod.get_pod_node(pod_obj).name for pod_obj in osd_pods]
+                common_node = list(set(mon_pod_nodes_name) & set(osd_pod_nodes_name))
+
+                if common_node:
+                    app_pod_nodes_name.append(common_node[0])
+                else:
+                    app_pod_nodes_name.append(osd_pod_nodes_name[0])
+
+                if num_of_nodes > 1:
+                    available_nodes = list(set(ocs_nodes) - set(mon_pod_nodes_name + osd_pod_nodes_name))
+                    app_pod_nodes_name += available_nodes[0:num_of_nodes - 1]
+
+        else:
+            app_pod_nodes_name += non_ocs_nodes[0:num_of_nodes]
+
+        return app_pod_nodes_name
+
+    def setup_pods_run_ios(self, pod_list, fio_filename='io_file'):
+        """
+        Setup pods and run ios.
+        Calculates md5sum of the data which can be used to verify data
+            integrity later
+
+        Args:
+            pod_list (list): list of pod objects to run ios
+
+        Returns:
+            md5sum_data (list): list of md5sum for the fio file
+        """
+        logger.info("Setting up pods for running IO")
+        executor = ThreadPoolExecutor(max_workers=len(pod_list))
+        for pod_obj in pod_list:
+            executor.submit(pod_obj.workload_setup, storage_type='fs')
+
+        # Wait for setup on pods to complete
+        for pod_obj in pod_list:
+            for sample in TimeoutSampler(
+                1000, 30, getattr, pod_obj, 'wl_setup_done'
+            ):
+                if sample:
+                    logger.info(
+                        f"Setup for running IO is completed on pod "
+                        f"{pod_obj.name}."
+                    )
+                    break
+        logger.info("Setup for running IO is completed on pods")
+
+        # Start IO on each pod
+        logger.info("Start IO on pods")
+        for pod_obj in pod_list:
+            pod_obj.run_io(
+                storage_type='fs', size='200M', runtime=10,
+                fio_filename=fio_filename
+            )
+        logger.info("IO started on all pods.")
+
+        # Check IO and calculate md5sum of files
+        md5sum_data = []
+        for pod_obj in pod_list:
+            pod.get_fio_rw_iops(pod_obj)
+            md5sum_data.append(pod.cal_md5sum(pod_obj=pod_obj, file_name=fio_filename))
+        return md5sum_data
+
+    def node_network_failure(self, node_list):
```

Comment:
any chance we have a function with such implementation in other tests?
If not, thinking it would be good to take this out of the test file to some commonplace

Reply:
ack, will move to `ocs_ci/ocs/node.py`